plugins/rapture/nexus-coreui-plugin/src/test/java/org/sonatype/nexus/coreui/RepositoryXOTest.groovy

```diff
@@ -0,0 +1,103 @@
+package org.sonatype.nexus.coreui
+
+import javax.validation.Validation
+import javax.validation.Validator
+
+import org.sonatype.nexus.validation.group.Create
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class RepositoryXOTest
+    extends Specification
+{
+
+  @Shared
+  Validator validator = Validation.buildDefaultValidatorFactory().validator
+
+  static List INVALID_CHARACTERS = ['_', '#', ',', 'A', '*', ' ', '#', '\'', '\\', '/', '?', '<', '>', '|', ' ', '\r',
+                                    '\n', '\t', ',', '.'].asImmutable()
```

Comment:
would it be simpler to have a set of valid chars instead of invalids?  /^[a-zA-z0-9._]+$/ ?